src/auth/KeyRing.cc

```diff
@@ -22,12 +22,21 @@
 #include "common/debug.h"
 #include "common/errno.h"
 #include "common/Formatter.h"
+#include "common/escape.h"
 
 #define dout_subsys ceph_subsys_auth
 
 #undef dout_prefix
 #define dout_prefix *_dout << "auth: "
 
+static string escape_str(const string& s)
+{
+   int len = escape_json_attr_len(s.c_str(), s.size());
+   char escaped[len];
+   escape_json_attr(s.c_str(), s.size(), escaped);
```

Comment:
i don't really think we should leverage `escape_json_attr_len()` and `escape_json_attr()` for escaping the cap line, even it works. because, please note, the cap line is not a json attribute.

what fails is how we re-create a config file without escaping the config value, and then feed it back to `ConfFile::parse_bufferlist()`. i think a minimal fix would be a utility function in `ConfLine`, so it is able to print key and escaped value like to given ostream, like
```c++
void print_escaped(std::ostream& os) {
  out << "\t" << key << " = \"" << escape(val) << std::endl;
}
```

Reply:
Thank you for the review. Will rework the fix following your suggestion.

Reply:
@tchaikov, looking into ConfFile, there is a problem in process_line(). It can not get the val with multiple quoted charachators. For example: 
caps mon = "allow r; allow command "auth del" with entity prefix client.manila."
ConfFile::process_line() needs some improvements to parse val: escape all quoted charactors except  the start and end ones?

Reply:
i don't think

```
caps mon = "allow r; allow command "auth del" with entity prefix client.manila."
```
is a valid cap setting.

could you prepare a test case?

Reply:
@tchaikov the following is a valid mon caps:
caps mon = "allow command "osd tree", allow command "osd df""
With this valid caps, we can run into the issue.
Here are the steps I run to test the issue:
$ceph auth get-or-create client.test mon 'allow command "osd tree", allow command "osd df"' -o /tmp/client.test.keyring
$ceph auth list output like this:
client.test
key: AQBqfJZarJBsBBAA3uWygbwovUC7rLGQClYbQQ==
caps: [mon] allow command "osd tree", allow command "osd df"
$ceph auth export client.test >/tmp/client.test.keyring

$cat /tmp/client.test.keyring
[client.test]
key = AQBqfJZarJBsBBAA3uWygbwovUC7rLGQClYbQQ==
caps mon = "allow command "osd tree", allow command "osd df""

$ ceph auth import -i /tmp/client.test.keyring
Error EINVAL: auth import: no caps supplied